test/ArrayIterator-test.js

```diff
@@ -679,4 +595,88 @@ describe('ArrayIterator', () => {
       });
     });
   });
+
+  describe('An ArrayIterator undefined iterator and autostart', () => {
+    let iterator;
+    before(() => {
+      iterator = new ArrayIterator(undefined, { autoStart: true });
```

Comment:
```suggestion
  describe('An ArrayIterator without source with autostart', () => {
    let iterator;
    before(() => {
      iterator = new ArrayIterator({ autoStart: true });
```

(should be made to work if it doesn't)